python/src/aac/plugins/check/check_aac_impl.py

```diff
@@ -137,8 +137,8 @@ def check_schema_constraint(
                     # if the field is a list, check each item in the list
                     if type(getattr(check_me, field.name)) != list:
                         raise LanguageError(
-                            f"Value of '{field.name}' was expected to be list, but was '{type(getattr(check_me, field.name))}'. File: '{source_definition.source.uri}'",
-                            None
+                            f"Value of '{field.name}' was expected to be list, but was '{type(getattr(check_me, field.name))}'",
+                            f'{source_definition.source.uri}'
```

Comment:
Wouldn't drop the additional file designator. Always best to be as explicit in returned errors as you can be.

```suggestion
                            f"Encountered in file: {source_definition.source.uri}"
```

Reply:
Thats what the __str__ method is for.  Its not neccessary to have a designator before the location because the __str__ has one, and when the error is diplayed it calls the __str__ method to display it.  It will show as 
"Error: {message}. Location: {location}"

Reply:
We can change the actual text of the __str__ method to be anything if we want to also.

Reply:
Okay, so that is part of what I was trying to get at when trying to understand why you implemented the formatting method how you did. I'll try to take a look for making sure this prints out like you expected. Because as it's leveraged right now, the `LanguageError` is just caught as Exceptions then the `message` and `location` portions are accessed from the exception. So, your formatting method won't really play a role as it stands.

Reply:
Is that something you are wanting to change with how the `LanguageError` is leveraged by pieces like `ExecutionMessage` to make use of the `__str__` formatting method you added?

Reply:
I don't think anything would have to change about ExecutionMessage, it doesnt directly interact with language error.  In some places, e.message and e.location are passed into an ExecutionMessage, and that will work as it does currently.

Reply:
So, I am only finding LanguageError to be caught then the individual pieces referenced for displaying the content. Because of that, I don't think your formatting method is ever going to be used. Have you seen different behavior somewhere that I am missing?

Reply:
When a language error is not caught and displays in the terminal, the __str__ method is called.

Reply:
Had side bar further discussing and reviewing the format method and its usage by the tool as backup provision when the exception is just raised and not caught for further control. Should be good to move on from this now.